packages/appium/lib/extension/extension-config.js

```diff
@@ -113,11 +124,94 @@ export class ExtensionConfig {
   }
 
   /**
+   * Retrieves listing data for extensions via command class.
+   * Caches the result in {@linkcode ExtensionConfig._listDataCache}
+   * @protected
+   * @returns {Promise<ExtensionListData>}
+   */
+  async getListData() {
+    if (this._listDataCache) {
+      return this._listDataCache;
+    }
+    const CommandClass = /** @type {ExtCommand<ExtType>} */ (commandClasses[this.extensionType]);
+    const cmd = new CommandClass({config: this, json: true});
+    const listData = await cmd.list({showInstalled: true, showUpdates: true});
+    this._listDataCache = listData;
+    return listData;
+  }
+
+  /**
+   * Displays non-"fatal" warnings for the extension
+   *
+   * This method uses Appium's logger, since the default extension logger uses the `error` log level.
+   *
    * @param {ExtManifest<ExtType>} extData
    * @param {ExtName<ExtType>} extName
+   * @returns {Promise<void>}
+   */
+  async displayConfigWarnings(extData, extName) {
```

Comment:
so are we going to be polling npm for possible package updates on every boot of the server? seems like it may be a waste of time just for letting the person know that there's an extension upgrade available.

Reply:
Only if the peer deps aren't satisfied.

As whether or not to do this, or just ask the user to check for updates themselves, I could go either way--both have benefits and drawbacks.

Reply:
ok. as long as it only happens in the case where peer deps aren't satisfied, i think that's worth it.